src/steps/big-table/index.ts

```diff
@@ -0,0 +1,320 @@
+import {
+  createDirectRelationship,
+  IntegrationStep,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+import { IntegrationConfig, IntegrationStepContext } from '../../types';
+import { BigTableClient } from './client';
+import {
+  ENTITY_CLASS_BIG_TABLE_APP_PROFILE,
+  ENTITY_CLASS_BIG_TABLE_BACKUP,
+  ENTITY_CLASS_BIG_TABLE_CLUSTER,
+  ENTITY_CLASS_BIG_TABLE_INSTANCE,
+  ENTITY_CLASS_BIG_TABLE_OPERATION,
+  ENTITY_CLASS_BIG_TABLE_TABLE,
+  ENTITY_TYPE_BIG_TABLE_APP_PROFILE,
+  ENTITY_TYPE_BIG_TABLE_BACKUP,
+  ENTITY_TYPE_BIG_TABLE_CLUSTER,
+  ENTITY_TYPE_BIG_TABLE_INSTANCE,
+  ENTITY_TYPE_BIG_TABLE_OPERATION,
+  ENTITY_TYPE_BIG_TABLE_TABLE,
+  RELATIONSHIP_TYPE_CLUSTER_HAS_BACKUP,
+  RELATIONSHIP_TYPE_INSTANCE_HAS_APP_PROFILE,
+  RELATIONSHIP_TYPE_INSTANCE_HAS_CLUSTER,
+  RELATIONSHIP_TYPE_INSTANCE_HAS_TABLE,
+  STEP_BIG_TABLE_APP_PROFILES,
+  STEP_BIG_TABLE_BACKUPS,
+  STEP_BIG_TABLE_CLUSTERS,
+  STEP_BIG_TABLE_INSTANCES,
+  STEP_BIG_TABLE_OPERATIONS,
+  STEP_BIG_TABLE_TABLES,
+} from './constants';
+import {
+  createAppProfileEntity,
+  createBackupEntity,
+  createClusterEntity,
+  createInstanceEntity,
+  createOperationEntity,
+  createTableEntity,
+} from './converters';
+
+export async function fetchOperations(
+  context: IntegrationStepContext,
+): Promise<void> {
+  const { instance, jobState } = context;
+  const client = new BigTableClient({ config: instance.config });
+
+  await client.iterateOperations(async (operation) => {
+    await jobState.addEntity(createOperationEntity(operation));
+  });
+}
+
+export async function fetchInstances(
+  context: IntegrationStepContext,
+): Promise<void> {
+  const { instance, jobState } = context;
+  const client = new BigTableClient({ config: instance.config });
+  const projectId = client.projectId;
```

Comment:
Apologies, I'm afraid I don't follow you. Are you referring to `projectId`?

Reply:
Yep, you could just use `client.projectId` directly (since it's only used once). :slightly_smiling_face: By the way, I'm not trying to make up vague change requests for the sake of making them :grinning: it's just that `client.projectId` is used a lot in the code base just like that, so this will 1) make the code more streamlined and 2) get rid of that unnecessary variable

Reply:
Oh it's not an issue haha. I can completely understand where you're coming from. I was just confused about what you were referring to. All good!